Fix globals, part 6: `performance` (#33339)
mdn/content: files/en-us/web/api/workerglobalscope/performance/index.md

---
title: "WorkerGlobalScope: performance property"
short-title: performance
slug: Web/API/WorkerGlobalScope/performance
page-type: web-api-instance-property
browser-compat: api.performance
---

{{APIRef("Performance API")}}

> The **`performance`** property of the {{domxref("WorkerGlobalScope")}} interface returns a {{domxref("Performance")}} object, which can be used to gather performance information about the context it is called in worker.

Reviewer: ```suggestion
The **`performance`** property of the {{domxref("WorkerGlobalScope")}} interface returns a {{domxref("Performance")}} object, which can be used to gather performance information about code running in the worker's scope.
```

> Performance entries are per context. If you create a mark on the main thread (or other worker), you cannot see it in a worker thread, and vice versa.

Reviewer: ```suggestion
Performance entries are per context. If you create a mark on a worker thread, you will not see it in the main thread or any other workers.
```

## Value

A {{domxref("Performance")}} object offering access to performance and timing-related information for the context it is called on worker.

## Performance API availability

> See the table which provides an overview about the availability of the performance APIs in window and worker contexts at {{domxref("Window.performance")}}.

Reviewer: We should not do this. I think we should scrap that table in the Window page, and instead list in this page the APIs that are available to workers, something like:

```
Note that only the following performance API are available in worker contexts:
...list available APIs
```

....and we should put it in the preamble, just before the "Value" H2.

---
title: "WorkerGlobalScope: performance property"
short-title: performance
slug: Web/API/WorkerGlobalScope/performance
page-type: web-api-instance-property
browser-compat: api.performance
---

{{APIRef("Performance API")}}

The **`performance`** property of the {{domxref("WorkerGlobalScope")}} interface returns a {{domxref("Performance")}} object, which can be used to gather performance information about code running in the worker's scope.

Performance entries are per context. If you create a mark on a worker thread, you will not see it in the main thread or any other workers.

Note that only the following performance interfaces are available in worker contexts:

- {{domxref("Performance")}}
- {{domxref("PerformanceEntry")}}
- {{domxref("PerformanceMark")}}
- {{domxref("PerformanceMeasure")}}
- {{domxref("PerformanceObserver")}}
- {{domxref("PerformanceObserverEntryList")}}
- {{domxref("PerformanceResourceTiming")}}
- {{domxref("PerformanceServerTiming")}}

## Value

> A {{domxref("Performance")}} object offering access to performance and timing-related information for the context it is called on worker.

Reviewer: ```suggestion
A {{domxref("Performance")}} object offering access to performance and timing-related information for the context it is called on.
```
